packages/commerce-sdk-react/src/query-client-config.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {DefaultOptions, QueryClientConfig} from '@tanstack/react-query'
+import {RetryValue} from '@tanstack/query-core/build/lib/retryer'
+
+const NUM_OF_RETRIES = 3 // same as React Query's default
+
+// NOTE: ResponseError does not seem to be exported from the isomorphic sdk lib, so I had to mimic it here
+interface ResponseError extends Error {
+    response: Response
+}
+
+const shouldContinueRetries: RetryValue<unknown> = (failureCount, error) => {
+    let shouldContinue = true
+
+    // See https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/blob/main/src/static/responseError.ts
+    const isResponseError = Boolean((error as ResponseError).response)
+
+    // ResponseError is more like unexpected runtime error, so we do want the retries.
```

Comment:
Should we retry based on the HTTP status code?

For example, don't retry on 4xx because these are client error (most likely validation errors from the server), but do retry on 500.

Reply:
Ah good idea.. ok let's retry only on 5xx.